cmd/utils/config.go

```diff
@@ -193,6 +193,28 @@ func SetP2PConfig(ctx *cli.Context, cfg *p2p.Config) {
 
 	cfg.NoDiscovery = ctx.Bool(NoDiscoverFlag.Name)
 
+	if ctx.IsSet(DiscoverTypesFlag.Name) {
+		nodetypes := strings.Split(ctx.String(DiscoverTypesFlag.Name), ",")
+		for _, nodetype := range nodetypes {
+			switch strings.ToLower(nodetype) {
+			case "auto":
```

Comment:
Can we remove `cfg.DiscoverTypes.Auto`, and determine my node type and set accordingly?

```
case "auto":
	if this is CN:
		cfg.DiscoverTypes.CN = true
	else if this is PN or EN:
		cfg.DiscoverTypes.PN = true
		cfg.DiscoverTypes.EN = true
```